module/AreaMaster.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file   AreaMaster.cpp
+ * @brief  エリアを攻略するクラス
+ * @author keiya121
+ */
+
+#include "AreaMaster.h"
+
+using namespace std;
+
+AreaMaster::AreaMaster(Area _area, bool _isLeftCourse, bool& _isLeftEdge, int _targetBrightness)
+  : area(_area),
+    isLeftCourse(_isLeftCourse),
+    isLeftEdge(_isLeftEdge),
+    targetBrightness(_targetBrightness)
+{
+}
+
+void AreaMaster::run()
+{
+  const int BUF_SIZE = 128;
+  char buf[BUF_SIZE];  // log用にメッセージを一時保持する領域
+  Logger logger;
+
+  // ファイルから受け取る動作リスト
+  vector<Motion*> motionList;
+
+  // コマンドファイルパスを作成する
+  char commandFilePath[BUF_SIZE];
+  snprintf(commandFilePath, BUF_SIZE, "%s%s%s.csv", basePath,
+           commandFileNames[static_cast<int>(area)], (isLeftCourse ? "Left" : "Right"));
+
+  // 動作インスタンスのリストを生成する
+  motionList = MotionParser::createMotions(commandFilePath, targetBrightness, isLeftEdge);
+
+  // 動作実行のメッセージログを出す
+  snprintf(buf, BUF_SIZE, "\nRun the commands in '%s'\n", commandFilePath);
+  logger.logHighlight(buf);
+
+  // 各動作を実行する
+  for(const auto& motion : motionList) {
+    motion->logRunning();
+    motion->run();
+  }
```

Comment:
【次回以降に回しても良いと思うが、みんなの意見も聞きたい部分】
ここに関してなんだけど、以下のようにした方が良いと思う
- run()を実行したオブジェクトを順次解放する
- motionListからもオブジェクトを順次削除する（こっちはそんなに影響ないかも？）

変更するなら以下のようにできる？のかな。
ローカルでビルドとテストは通ったけど、実機で走るかは確認してません。
普通に間違ってたら指摘お願いします。
```suggestion
  // 各動作を実行する
  for(auto motion = motionList.begin(); motion != motionList.end(); ) {
    (*motion)->logRunning();
    (*motion)->run();
    delete *motion;  // メモリを解放
    motion = motionList.erase(motion);  // リストから削除
  }
```

Reply:
run()したオブジェクトを残しておく必要は無さそうので、そちらの方が良さそうと自分も思いました。
実機で動くかも検証してみます。

Reply:
https://github.com/KatLab-MiyazakiUniv/etrobocon2024/assets/107525125/9a440876-d209-45fe-86d7-75a77444dfc7

テストのため、csvファイルの回頭動作のコマンドを以下の3つにして実行してみましたが、問題無く動作しました。

```
AR,90,100,clockwise,指定角度回頭
AR,90,100,anticlockwise,指定角度回頭
AR,90,100,clockwise,指定角度回頭
```

Reply:
ありがとう！